include/domain-types.h

```diff
@@ -16,14 +17,73 @@ namespace connect
 
 typedef int32_t id_t;
 
-typedef int status_t;
-
-enum
+class Status
 {
-    STATUS_OK       = 0,
-    STATUS_ERROR    = 1
+public:
+    Status(int code, bool isError, int facility)
+    {
+        assert(0 <= code  &&  code <= MAX_CODE);
+        assert(0 <= facility  &&  facility <= MAX_FACILITY);
+
+        status_ = (isError ? FAIL_BIT : 0) | ((uint32_t)facility << 16) | (uint32_t)code;
+    }
+
+    // codes
+    enum
+    {
+        SUCCESS = 0,
+        FAILURE = 1, // for any (unknown) reason
+        MAX_CODE = 0xffff // almost 64k values range
+    };
+
+    enum
+    {
+        FACILITY_NONE = 0, // for general codes
+        FACILITY_NETWORK = 1,
+        FACILITY_WEBAPI = 2,
+        MAX_FACILITY = 0x7ff // ~2000 values range
+    };
+
+    int getCode() const
+    {
+        return status_ & CODE_MASK;
+    }
+
+    int getFacility() const
+    {
+        return (status_ >> 16) & FACILITY_MASK;
+    }
+
+    bool isError() const
```

Comment:
If class was Error rather, than Status, it could be. Currently is doesn't have meaning: if (status) or if (!status) what should it mean? status.isSuccess() and status.isError() is clear